classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,408 @@
+
+    /**
+     * Set context issues filter
+     *
+     * @return $this
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Set result order column
+     *
+     * @return $this
+     */
+    public function orderBy(string $column): self
+    {
+        switch ($column) {
+            case self::ORDERBY_LAST_MODIFIED:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'i.last_modified',
+                        'direction' => self::ORDER_DIR_DESC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_SEQUENCE:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'o.seq',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_PUBLISHED_ISSUES:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'o.seq',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'currentIssue',
+                        'direction' => self::ORDER_DIR_DESC
+                    ],
+                    [
+                        'orderBy' => 'i.date_published',
+                        'direction' => self::ORDER_DIR_DESC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_UNPUBLISHED_ISSUES:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'i.year',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.volume',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.number',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_SHELF:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => self::ORDER_CURRENT_ISSUE,
+                        'direction' => self::ORDER_DIR_DESC
+                    ],
+                    [
+                        'orderBy' => 'i.year',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.volume',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.number',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            default:
```

Comment:
Rather than having a default that's always accepted, I'd suggest making all the valid calls to this function defined constants, and having `default: throw new \InvalidArgumentException(...)`